src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -58,112 +80,88 @@ var canUseCollections = (
   isNative(Set.prototype.keys)
 );
 
-var itemMap;
-var rootIDSet;
+if (canUseCollections) {
+  var itemMap = new Map();
+  var rootIDSet = new Set();
 
-var itemByKey;
-var rootByKey;
+  var setItem = function(id, item) {
+    itemMap.set(id, item);
+  };
+  var getItem = function(id) {
+    return itemMap.get(id);
+  };
+  var removeItem = function(id) {
+    itemMap.delete(id);
+  };
+  var getItemIDs = function() {
+    return Array.from(itemMap.keys());
+  };
 
-if (canUseCollections) {
-  itemMap = new Map();
-  rootIDSet = new Set();
-} else {
-  itemByKey = {};
-  rootByKey = {};
-}
+  var addRoot = function(id) {
+    rootIDSet.add(id);
+  };
+  var removeRoot = function(id) {
+    rootIDSet.delete(id);
+  };
+  var getRootIDs = function() {
+    return Array.from(rootIDSet.keys());
+  };
 
-var unmountedIDs = [];
+} else {
+  var itemByKey = {};
+  var rootByKey = {};
 
-// Use non-numeric keys to prevent V8 performance issues:
-// https://github.com/facebook/react/pull/7232
-function getKeyFromID(id) {
-  return '.' + id;
-}
-function getIDFromKey(key) {
-  return parseInt(key.substr(1), 10);
-}
+  // Use non-numeric keys to prevent V8 performance issues:
+  // https://github.com/facebook/react/pull/7232
+  function getKeyFromID(id: DebugID): string {
+    return '.' + id;
+  }
+  function getIDFromKey(key: string): DebugID {
+    return parseInt(key.substr(1), 10);
+  }
```

Comment:
arg, I ran npm test but didn't run the lint. Travis is soo backed up, I'll send a quick fix